ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/date/DateTaskFragment.kt

```diff
@@ -20,26 +20,36 @@ import android.text.format.DateFormat
 import android.view.LayoutInflater
 import android.view.View
 import com.google.android.ground.databinding.DateTaskFragBinding
+import com.google.android.ground.model.submission.DateTimeTaskData
+import com.google.android.ground.model.submission.isNotNullOrEmpty
 import com.google.android.ground.ui.datacollection.components.TaskView
 import com.google.android.ground.ui.datacollection.components.TaskViewFactory
 import com.google.android.ground.ui.datacollection.tasks.AbstractTaskFragment
 import dagger.hilt.android.AndroidEntryPoint
 import java.util.*
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
 import org.jetbrains.annotations.TestOnly
 
 @AndroidEntryPoint
 class DateTaskFragment : AbstractTaskFragment<DateTaskViewModel>() {
 
   private var datePickerDialog: DatePickerDialog? = null
 
+  private var _dateText: MutableStateFlow<String> = MutableStateFlow("")
+  val dateText = _dateText.asStateFlow()
```

Comment:
Can we chain the task data to date conversion here directly instead of calling it manually from the below two places?